src/sentry/api/endpoints/organization_pinned_searches.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers
+from rest_framework.response import Response
+from django.utils import six
+
+from sentry.api.bases.organization import (
+    OrganizationEndpoint,
+    OrganizationPinnedSearchPermission,
+)
+from sentry.api.serializers import serialize
+from sentry.models import SavedSearch
+from sentry.models.search_common import SearchType
+
+
+class OrganizationSearchSerializer(serializers.Serializer):
+    type = serializers.IntegerField(required=True)
+    query = serializers.CharField(required=True)
+
+    def validate_type(self, attrs, source):
+        try:
+            SearchType(attrs[source])
+        except ValueError as e:
+            raise serializers.ValidationError(six.text_type(e))
+        return attrs
+
+
+class OrganizationPinnedSearchEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationPinnedSearchPermission, )
+
+    def post(self, request, organization):
```

Comment:
Should this be PUT, as the action create or overwrites? :bike::house:

Reply:
Yeah, maybe that's more correct. Did some reading and one thing is that `PUT` requests should be idempotent, which is what will happen here, since pinning the same thing won't result in any changes.

Reply:
Done, cc @billyvg, we're changing the pin endpoint to be `PUT` rather than `POST`